pallets/nbv-storage/src/tests.rs

```diff
@@ -0,0 +1,126 @@
+
+use crate::{mock::*, Error};
+use pallet_identity::{IdentityInfo, Data};
+use codec::{Encode};
+use frame_support::{
+	assert_noop, assert_ok,
+	traits::{ConstU32,},
+	BoundedVec,
+};
+
+fn dummy_identity(name : &[u8] ) -> IdentityInfo<MaxAdditionalFields> {
+	IdentityInfo {
+		display: Data::Raw(name.to_vec().try_into().unwrap()),
+		legal: Data::Raw(b"The Right Ordinal Ten, Esq.".to_vec().try_into().unwrap()),
+		web: Data::None,
+		riot: Data::None,
+		email: Data::None,
+		image: Data::None,
+		twitter: Data::None,
+		additional: Default::default(),
+		pgp_fingerprint: Default::default(),
+	}
+}
+
+fn dummy_wrong_identity() -> IdentityInfo<MaxAdditionalFields> {
+	let key = BoundedVec::<u8,ConstU32<32> >::try_from(b"xpub".encode())
+				.expect("Error on encoding the xpub key to BoundedVec");
+	let mut additional_info = BoundedVec::<(Data, Data),MaxAdditionalFields >::default();
+	additional_info.try_push( (Data::Raw(key), Data::None) )
+		.expect("Error pushing additional info while building dummy info");
+	IdentityInfo {
+		display: Data::Raw(b"ten".to_vec().try_into().unwrap()),
+		legal: Data::Raw(b"The Right Ordinal Ten, Esq.".to_vec().try_into().unwrap()),
+		web: Data::None,
+		riot: Data::None,
+		email: Data::None,
+		image: Data::None,
+		twitter: Data::None,
+		additional: additional_info,
+		pgp_fingerprint: Default::default(),
+	}
+}
+
+fn dummy_psbt() -> BoundedVec<u8, PSBTMaxLen> {
```

Comment:
This may be enough of the xpub for the app; perhaps it depends on the `derivation path` assumptions made in BDK. Some xpubs show a fingerprint and the derivation path.  

Here's an example from the `bdk-cli`:

```sh
bdk-cli key derive --xprv tprv8ZgxMBicQKsPdp21UgwHvn6ayoSbctppbD2SwChwNzkXLQKf6fYb6Kjtfww2muuTuZLRVpEV1Y4UbJYdwRnxBb1EPeqxGZmRbSo5NLXYWxc --path "m/84'/1'/0'/0"
```

It produces this output: 
```json
{
  "xprv": "[621c051d/84'/1'/0'/0]tprv8hnMza7pP8tvATh7qb2eUdxwuekBTtueSJFZkji9pJpu2LqdrT6ijHqe87SE2hXkt4Q6aU3JaTKTjLvVX9K4gupjJFN2WFs8sLafYNQySf5/*",
  "xpub": "[621c051d/84'/1'/0'/0]tpubDEUQ8zA4XWab3viujEhEt3d4UgG7dE6Z1brM3FkTEadHrq6QUqvJunTWJH6suxgGsFqspT9AMhASFTj531BVb145NKL2ZzyYGDNQaF2733U/*"
}
```
I don't know if the backend will expect the string beginning `xpub6A...` or it will expect the string beginning `[621c05...`

@sebastianmontero - do you know? 

@amatsonkali the fingerprint is always 8 bytes. I guess the derivation path could vary if the `/0` values go to `/10` or `/100`.  I guess that this can be handled by the XPubLen constant, but the UI <-> pallet <-> bdk services will need to all be aligned on the structure of the `xpub` string